storage/kafka_storage/kafka_storage.go

```diff
@@ -127,7 +129,8 @@ func (ks *KafkaStorage) Send(messages ...storage.Message) error {
 }
 
 func (ks *KafkaStorage) GetMessages(_ uint64) ([]storage.Message, error) {
-	ctx, _ := context.WithDeadline(context.Background(), time.Now().Add(time.Second*10))
+	ctx, cancel := context.WithDeadline(ks.readerCtx, time.Now().Add(time.Second*10))
```

Comment:
Adding hard-coded timeout constants is a bad practice. Please move this parameter to config.